c10/cuda/CUDAMallocAsyncAllocator.cpp

```diff
@@ -762,30 +774,15 @@ struct CudaMallocAsyncAllocator : public CUDAAllocator {
     }
 
     capture_free_streams.clear();
-  }
-
-  void notifyCaptureEnded(int device, CaptureId_t graph_id) override {
-    assertValidDevice(device);
-
-    std::lock_guard<std::mutex> lk(general_mutex);
-
-    TORCH_CHECK(
+      TORCH_CHECK(
         capture_underway,
         "CudaMallocAsync::notifyCaptureEnded called, "
         "but CudaMallocAsync::capture_underway is false.");
     capture_underway = false;
-
-    // See Note [Avoid freeing uncaptured ptrs during CUDA graph capture]
-    for (const auto ptr : ungraphed_ptrs_defer_free_until_no_capture) {
-      auto it = ptr_info.find(ptr);
-      TORCH_INTERNAL_ASSERT(it != ptr_info.end(), "ptr not found in ptr_info");
-      free_impl(it);
-    }
-
-    ungraphed_ptrs_defer_free_until_no_capture.clear();
```

Comment:
is this logic not important ?

Reply:
I moved it to `mallocAsync`: in other words once we are done graph capturing but before we do further allocations we free these buffers. It will end up being very similar to what happened before but allows us to have a more meaningful API for what is happening with pools (allocator knows about the intent to create and allocate into pools but doesn't need to know the specifics of the cuda graph API).